CLAUDE.md

```diff
@@ -0,0 +1,194 @@
+# CLAUDE.md
+
+This file provides guidance to Claude Code (claude.ai/code) when working with code in this repository.
+
+**For contribution guidelines, code standards, and additional development information not covered here, please refer to [CONTRIBUTING.md](./CONTRIBUTING.md).**
+
+## Repository Overview
+
+MLflow is an open-source platform for managing the end-to-end machine learning lifecycle. It provides tools for:
+
+- Experiment tracking
+- Model versioning and deployment
+- LLM observability and tracing
+- Model evaluation
+- Prompt management
+
+## Quick Start: Development Server
+
+### Start the Full Development Environment (Recommended)
+
+```bash
+# Kill any existing servers
+pkill -f "mlflow server" || true; pkill -f "yarn start" || true
+
+# Start both MLflow backend and React frontend dev servers
+nohup uv run bash dev/run-dev-server.sh > /tmp/mlflow-dev-server.log 2>&1 &
+
+# Monitor the logs
+tail -f /tmp/mlflow-dev-server.log
+
+# Servers will be available at:
+# - MLflow backend: http://localhost:5000
+# - React frontend: http://localhost:3000
+```
+
+This uses `uv` (fast Python package manager) to automatically manage dependencies and run the development environment.
+
+## Development Commands
+
+### Testing
+
+```bash
+# Run Python tests
+uv run pytest tests/
+
+# Run specific test file
+uv run pytest tests/test_version.py
```

Comment:
This currently doesn't work because our `pyproject.toml` doesn't have pytest?

Reply:
I changed this so that uv will install the pip requirements from test and use that version, long-term we should move all those requirements.txt files into dependency groups in pyproject.

Is it ok to keep UV here for now? It all works and is consistent, I've been struggling with getting dependencies to be consistent here without it (my guess is contributors would too)

Reply:
sounds good :)